test/test-applications/integrations/TestApplication.HttpServer/Program.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="Program.cs" company="Splunk Inc.">
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using Microsoft.AspNetCore.Builder;
+using TestApplication.Shared;
+
+namespace TestApplication.HttpServer;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        ConsoleHelper.WriteSplashScreen(args);
+
+        var builder = WebApplication.CreateBuilder(args);
+
+        var requestPath = "/request";
+        var app = builder.Build();
+        using var observer = new LifetimeObserver(app, requestPath);
```

Comment:
can't we provide a `/shutdown` endpoint instead? It should be faster and more readable.

Reply:
I would really want to keep it inproc, otherwise local testing will leave lot of stray processes. 
... and I'm not sure it will be faster? Currently shutdown will be called when `/request` serves response and appropriate activity is created, or the process is left hanging for 2 mins.
When shutdown control is out of process, there is additional overhead to create the request, which consumes probably more time.

Reply:
Makes sense 👍